crud.py

```diff
@@ -0,0 +1,83 @@
+from sqlalchemy.orm import Session
+
+import models
+import schemas
+
+
+def get_all_authors(
+    db: Session,
+    skip: int = 0,
+    limit: int = 5,
+):
+    return (
+        db.query(models.DBAuthor)
+        .offset(skip).limit(limit)
+        .all()
+    )
+
+
+def get_author(db: Session, author_id: int):
+    return (
+        db.query(models.DBAuthor)
+        .filter(models.DBAuthor.id == author_id)
+        .first()
+    )
+
+
+def get_author_by_name(db: Session, author_name: str):
+    return (
+        db.query(models.DBAuthor)
+        .filter(models.DBAuthor.name == author_name)
+        .first()
+    )
+
+
+def create_author(db: Session, author: schemas.AuthorCreate):
+    db_author = models.DBAuthor(
+        name=author.name,
+        bio=author.bio,
+    )
+    db.add(db_author)
+    db.commit()
+    db.refresh(db_author)
+
+    return db_author
+
+
+def get_all_books(
+    db: Session,
+    author_id: int | None = None,
+    skip: int = 0,
+    limit: int = 5,
+):
+    queryset = db.query(models.DBBook)
+
+    if author_id:
+        queryset = (
+            queryset
+            .filter(models.DBBook.author_id.has(author_id=author_id))
+        )
+
+    return queryset.offset(skip).limit(limit).all()
+
+
+def get_book_by_title(db: Session, book_title: str):
+    return (
+        db.query(models.DBBook)
+        .filter(models.DBBook == book_title)
```

Comment:
The filter condition `.filter(models.DBBook == book_title)` is incorrect. It should be `.filter(models.DBBook.title == book_title)` to properly filter books by their title.